pkg/api/dashboard.go

```diff
@@ -195,9 +196,14 @@ func deleteDashboard(c *models.ReqContext) Response {
 	}
 
 	err := dashboards.NewService().DeleteDashboard(dash.Id, c.OrgId)
-	if err == models.ErrDashboardCannotDeleteProvisionedDashboard {
-		return Error(400, "Dashboard cannot be deleted because it was provisioned", err)
-	} else if err != nil {
+	if err != nil {
+		var dashboardErr models.DashboardErr
+		if ok := errors.As(err, &dashboardErr); ok {
+			if errors.Is(err, models.ErrDashboardCannotDeleteProvisionedDashboard) {
```

Comment:
On second look, I think that the first condition (`errors.As`) is redundant here. The 2nd one should be enough.

Reply:
@papagian I thought the same for a second, but it's actually necessary to get the status code (`dashboardErr.StatusCode`).

Reply:
yes, I see now the problem.